consul/kv_consul.go

```diff
@@ -654,6 +654,12 @@ func (kv *consulKV) Snapshot(prefix string) (kvdb.Kvdb, uint64, error) {
 			} else {
 				if kvp.Action == kvdb.KVDelete {
 					_, err = snapDb.Delete(kvp.Key)
+					// A Delete key was issued between our Watch and Enumerate
```

Comment:
oh yes. will fix this

Reply:
I take my words back. This will happen only if a Delete on actual kvdb happens between our Watch and Enumerate APIs. Which Put are you talking about? If it happens between Watch and Enumerate then the Enumerate API will not return this deleted key and hence we won't SnapPut this key, however as the Watch was started before this Enumerate, we get a callback for Delete and we end up deleting a non-existent key. Is my understanding correct ?

Reply:
snapshot works by starting a watch (at any index), putting a key which is the lowest version at which watch starts, then enumerate and create snap db, then do another put to get highest version for watch stop. 
then we replay all the updates between the lowest and highest version which we got with put.
the bug is that if a key was deleted after the first put and before enumerate, watch returns an update for a key not in db.

Reply:
ah! I missed the fact that we only apply updates between the low and high index. Will fix the comment.